pkg/domain/workflow.go

```diff
@@ -10,20 +10,25 @@ import (
 // WorkflowStore is an inteface to workflow stores
 type WorkflowStore interface {
 	GetWorkflow(ctx context.Context, name string) *workflow.Workflow
-	GetAllWorkflows(ctx context.Context, name *string) (result []*workflow.Workflow)
+	GetWorkflows(ctx context.Context, name *string) (result []*workflow.Workflow)
 	AddWorkflow(ctx context.Context, w *workflow.Workflow) (*workflow.Workflow, error)
-	GetAssignedCodesets(ctx context.Context, workflowName string) []*Codeset
-	GetAssignments(ctx context.Context, workflowName *string) map[string][]*Codeset
-	AddCodesetAssignment(ctx context.Context, workflowName string, codeset *Codeset) []*Codeset
+	DeleteWorkflow(ctx context.Context, name string) error
+	GetAssignedCodeset(ctx context.Context, workflowName string, codeset *Codeset) *AssignedCodeset
+	GetAssignedCodesets(ctx context.Context, workflowName string) []*AssignedCodeset
+	GetAssignments(ctx context.Context, workflowName *string) map[string][]*AssignedCodeset
+	AddCodesetAssignment(ctx context.Context, workflowName string, assignedCodeset *AssignedCodeset) []*AssignedCodeset
+	DeleteCodesetAssignment(ctx context.Context, workflowName string, codeset *Codeset) []*AssignedCodeset
 }
 
 // WorkflowBackend is the interface for the FuseML workflows
 type WorkflowBackend interface {
-	CreateWorkflow(context.Context, *log.Logger, *workflow.Workflow) error
-	CreateWorkflowRun(context.Context, string, *Codeset) error
-	ListWorkflowRuns(context.Context, workflow.Workflow, WorkflowRunFilter) ([]*workflow.WorkflowRun, error)
-	CreateWorkflowListener(context.Context, *log.Logger, string, bool) (*WorkflowListener, error)
-	GetWorkflowListener(context.Context, *log.Logger, string) (*WorkflowListener, error)
+	CreateWorkflow(ctx context.Context, logger *log.Logger, workflow *workflow.Workflow) error
+	DeleteWorkflow(ctx context.Context, logger *log.Logger, workflowName string) error
+	CreateWorkflowRun(ctx context.Context, logger *log.Logger, workflowName string, codeset *Codeset) error
+	ListWorkflowRuns(ctx context.Context, workflow workflow.Workflow, filter WorkflowRunFilter) ([]*workflow.WorkflowRun, error)
+	CreateWorkflowListener(ctx context.Context, logger *log.Logger, workflowName string, wait bool) (*WorkflowListener, error)
+	DeleteWorkflowListener(ctx context.Context, logger *log.Logger, workflowName string) error
+	GetWorkflowListener(ctx context.Context, workflowName string) (*WorkflowListener, error)
```

Comment:
nit for later improvement: having to pass the logger to all these methods is a real PITA, given that it's always the same logger and it's every time you call one of them. Better modeled as a workflow backend member/constructor argument, don't you think ?